src/List.js

```diff
@@ -77,77 +129,79 @@ export function List() {
     const timeToMinDate = new Date() - minDate + ONE_MINUTE;
     //re-render the page so the item unchecks when it should be unchecked
     setTimeout(() => setReRender({}), timeToMinDate);
-  }, [reRender, items]);
+  }, [reRender, items, ONE_MINUTE]);
 
   const handleChange = async (id, event) => {
     let date = new Date();
     const item = items.find((element) => element.id === id);
     const daysSinceLastTransaction = item?.lastPurchasedDate
       ? convertToDays(Math.round(new Date() - item.lastPurchasedDate))
-      : 0;
+      : convertToDays(Math.round(new Date() - item.creationTime));
```

Comment:
No.  With `0` it has an issue, because the `daysSinceLastTransaction`  needs the number of days since the item was added to the list if it doesn't have `lastPurchasedDate`. Even in the `calculateEstimate` method there is this comment:
```
daysSinceLastTransaction: number, // **The number of days since the item was added to the list** or last purchased
```
 So that is also why we added the new attribute `creationItem`.

Reply:
sorry i was thinking more about if item is undefined, maybe !!item would work instead of item?.purchaseDate?  otherwise, with the current code you could just do Math.round(daysSinceLastPurchaseOrCreationTime(item)) for the whole expression